### PR TITLE
feat(my collection): present disambiguating info when selecting an artist

### PR DESCRIPTION
**Pull request by anandaroop**
> [!IMPORTANT]  
> As a reminder, **any** time we present a list of artists from which a user will select, we should also display disambiguating info: nationality, birth date, death date.

This PR resolves [ONYX-231] 

### Description

This addresses a long-standing issue in MyC: we do not consistently display disambiguating info about artists, which increases the risk of misattributed artworks and duplicative artist data entering the platform.

If you're thinking there's already disambiguating info in the **Before** screenshots below — that's true, but only because people have abused one of the other name fields, presumably in response to ambiguous displays somewhere in the product:

The better we are about displaying the disambiguating info, the less the chance of that happening.

### Adding a collected artist to MyC

|Before|After (iOS)|After (Android)|
|---|---|---|
||||

### Adding an artwork to MyC

|Before|After (iOS)|After (Android)|
|---|---|---|
||||

Note that the gist of this PR is in the first three commits. I also tried to be a good citizen 😇 and address a bunch of linting errors and deprecations in the existing tests, that's the last two commits.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Artist search results present disambiguating information - anandaroop

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-231]: https://artsyproduct.atlassian.net/browse/ONYX-231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Artist search results present disambiguating information - anandaroop)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c91b53c869c2efb4e19ccd3d5b0aa49f28d3e909
</p>

File: src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtistsAutosuggest.tsx [hunk 1]
```diff
@@ -245,6 +246,7 @@ const CollectedArtistListItem: React.FC<{
 
         <ResultWithHighlight
           displayLabel={artist.displayLabel!}
+          secondaryLabel={artist.formattedNationalityAndBirthday || nbsp}
```
Comment (anandaroop) [inline src/app/Scenes/MyCollection/Screens/MyCollectionAddCollectedArtists/MyCollectionAddCollectedArtistsAutosuggest.tsx:249]: Value or else `nbsp` to ensure proper horizontal alignment when disambiguating info is _not_ available:

File: src/app/Scenes/Search/components/AutosuggestSearchResult.tsx [hunk 2]
```diff
@@ -110,7 +110,12 @@ export const AutosuggestSearchResult: React.FC<{
 
   const resultType = getResultType(result)
 
-  const initials = result.__typename === "Artist" ? result.initials : undefined
+  let initials, secondaryLabel
+
+  if (result.__typename === "Artist") {
+    initials = result.initials
+    secondaryLabel = result.formattedNationalityAndBirthday || undefined
```
Comment (anandaroop) [inline src/app/Scenes/Search/components/AutosuggestSearchResult.tsx:117]: This `formattedNationalityAndBirthday` was already being requested in the corresponding Relay query, we just needed to actually make use of it.

File: src/app/Scenes/Search/components/ResultWithHighlight.tsx [hunk 3]
```diff
@@ -4,25 +4,40 @@ import parse from "autosuggest-highlight/parse"
...
+
+      {!!secondaryLabel && (
         <Text
-          key={i}
           variant="xs"
-          color={part.highlight ? "blue100" : "black100"}
-          weight={part.highlight ? "medium" : "regular"}
+          numberOfLines={1}
+          textAlign={textAlign}
+          ellipsizeMode="middle"
```
Comment (anandaroop) [inline src/app/Scenes/Search/components/ResultWithHighlight.tsx:35]: For the edge case of a veeerrryy long `formattedNationalityAndBirthday` field, I've chosen to truncate from the middle. I'm open to other ideas.

File: src/app/Scenes/Search/components/AutosuggestSearchResult.tests.tsx [hunk 4]
```diff
@@ -1,11 +1,10 @@
 import { Touchable } from "@artsy/palette-mobile"
-import { fireEvent } from "@testing-library/react-native"
+import { fireEvent, screen } from "@testing-library/react-native"
 import { SearchContext } from "app/Scenes/Search/SearchContext"
 import { GlobalStore, GlobalStoreProvider } from "app/store/GlobalStore"
 import { EntityType, navigate, navigateToEntity, SlugType } from "app/system/navigation/navigate"
 import { CatchErrors } from "app/utils/CatchErrors"
-import { extractText } from "app/utils/tests/extractText"
-import { renderWithWrappers, renderWithWrappersLEGACY } from "app/utils/tests/renderWithWrappers"
```
Comment (anandaroop) [inline src/app/Scenes/Search/components/AutosuggestSearchResult.tests.tsx:8]: 👋🏽 😅

**Review by nickskalkin — APPROVED** [review]
👏

**Review by olerichter00 — APPROVED** [review]
Looks great 🌟

**Review by dblandin — APPROVED** [review]
Looks fantastic! 💯

File: src/app/Scenes/Search/components/AutosuggestSearchResult.tsx (hunk 2 above)
Comment (mzikherman) [reply src/app/Scenes/Search/components/AutosuggestSearchResult.tsx:117]: (i wonder if there's some easy tooling around checking if we're requesting fields in fragments that _aren't_ being used in a component)

**Comment by mzikherman** [comment]
Great idea!
